Sources/SPTDataLoader.m

```diff
@@ -175,10 +198,19 @@ - (void)successfulResponse:(SPTDataLoaderResponse *)response
     if (![self isRequestExpected:response.request]) {
         return;
     }
-    
-    [self executeDelegateBlock: ^{
-        [self.delegate dataLoader:self didReceiveSuccessfulResponse:response];
-    }];
+    if (self.delegate) {
+        [self executeDelegateBlock:^{
+            [self.delegate dataLoader:self didReceiveSuccessfulResponse:response];
+        }];
+    } else {
```

Comment:
Handlers should probably be resolved regardless of whether or not a delegate is set

Reply:
That sounds reasonable.
A check before calling `executeDelegateBlock` can made to see if we have any handlers to execute

Reply:
Does this mean both delegate and handler must be invoked, when both exist?

Reply:
Usually my rule of thumb is "if you ask for weird things you get weird things" I would not expect a client to set both a delegate and block but I'd rather enforce them at API level to not be possible. If we allow them to be set both so the expectation is that both are being called.
The answer to your question, yes, if both exist then both has to be called.